internal/storage/v2/elasticsearch/tracestore/writer.go

```diff
@@ -0,0 +1,43 @@
+// Copyright (c) 2025 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+package tracestore
+
+import (
+	"context"
+
+	"go.opentelemetry.io/collector/pdata/ptrace"
+
+	"github.com/jaegertracing/jaeger-idl/model/v1"
+	cfg "github.com/jaegertracing/jaeger/internal/storage/elasticsearch/config"
+	"github.com/jaegertracing/jaeger/internal/storage/v1/elasticsearch/spanstore"
+)
+
+type TraceWriter struct {
+	spanWriter spanstore.CoreSpanWriter
+}
+
+// NewTraceWriter returns the TraceWriter for use
+func NewTraceWriter(p spanstore.SpanWriterParams) *TraceWriter {
+	return &TraceWriter{
+		spanWriter: spanstore.NewSpanWriter(p),
+	}
+}
+
+// WriteTraces convert the traces to ES Span model and write into the database
+func (t *TraceWriter) WriteTraces(_ context.Context, td ptrace.Traces) error {
+	dbSpans := ToDBModel(td)
+	for i := 0; i < len(dbSpans); i++ {
+		span := &dbSpans[i]
+		t.spanWriter.WriteSpan(model.EpochMicrosecondsAsTime(span.StartTime), span)
```

Comment:
I think this is where we should start diverging with v1 implementation. One of the strengths of v2 storage API is that it allows batch writes. v1 API was a single span at a time. So v1 writer had a workaround by running a background batch insert, which also had a problem of not being able to report errors from the writes. 

I think we can do better in v2. I don't know if we even need to have a background batching, because this can already configured in the pipeline above the storage. Instead we can treat each payload as already large-enough batch and write all spans in a single request.